### ISSUE TITLE
Images not getting displayed (img tag)

### ISSUE THREAD

**Issue by jeevatkm** [tdewolff/minify#71]
I'm using minify writer for minifying my app HTML. It does minify of HTML, inline CSS and & JS. I'm facing an issue of images are not displayed in the browser.

BTW, why double quotes are getting striped from html attributes?

**Comment by tdewolff** [comment]
Because quotes around HTML attributes are optional and thus removed by the minifier.

Can you show the shortest example that demonstrates this problem? What is the input and output of that gives this problem and what browser and OS are you using?

**Comment by jeevatkm** [comment]
I'm using Mac OS X - Yosemite. I have verified with Safari, Firefox, Chrome. Screenshot attached.

Input is `http.ResponseWriter`:
```go
// m is minify instance, created like this m = minify.New()
mw := m.Writer(mimeHTML, c.Response.Out)
writer := MinifyWriter{c.Response.Out, mw, mimeHTML, false}
c.Response.Out = &writer
```

Output Image Tag for this:
```html
<a href=# class=dropdown-toggle data-toggle=dropdown style=padding-top:14px;padding-bottom:15px;padding-left:10px><img src=/public/img/profile-pic-32x32.png class=img-circle id=userProfileImage></a>
```
<img width="243" alt="screen shot 2016-01-25 at 11 42 04 am" src="https://cloud.githubusercontent.com/assets/1175977/12562246/0cd39f5e-c359-11e5-8009-0eb726b48236.png">

Output Image Tag for this:
```html
<div class=user-pic><img src=/public/img/profile-pic-128x128.png class=img-circle alt="User Pic"></div>
```
<img width="269" alt="screen shot 2016-01-25 at 11 41 55 am" src="https://cloud.githubusercontent.com/assets/1175977/12562247/0cf08358-c359-11e5-86db-f04fe41bdd96.png">

**Comment by jeevatkm** [comment]
BTW, if I do like this. Then minify output is not as excepted/efficient but images getting displayed properly.
```go
m.Add("text/html", &html.Minifier{
 	KeepDefaultAttrVals: true,
 	KeepWhitespace:      true,
 })
```

**Comment by tdewolff** [comment]
What is the input HTML? Can you try by manually adding quotes around the `src` attribute if that resolves this issue?

**Comment by jeevatkm** [comment]
Img tag has double quotes in the input. I'm debugging and working on it. Additional info, static files is getting delivered by nginx not by Go.

Can I have your email Id? I can send HTML output with minify and without minify. I cannot post it github.

**Comment by jeevatkm** [comment]
@tdewolff Finally found the root cause. Minify writer and revel framework response filter chain was not integrated well. That was messing up image content.

Now I have successfully integrated Minify library with Revel framework :)

Thank so much for your time and prompt response.

**Comment by tdewolff** [comment]
What was the problem? And how was it fixed? Or was it something outside Minify that caused the problem? It might help other folks in the future.

**Comment by jeevatkm** [comment]
There was two issues, even though I supply minify writer and implement Write method.
* Minify was applied to non-html mime types too (e.g. image)
* Randomly content length mismatch

Finally I referred [compress.go](https://github.com/revel/revel/blob/master/compress.go) from revel framework re-written logic for minify. Now it gets applied only to HTML, inline CSS & JS. I have sent code via email. Thinking to write an article in my blog, since I couldn't find a reference when I searched around.
